ACE/include/makeinclude/platform_macosx_iOS.GNU

```diff
@@ -10,15 +10,24 @@ else
 endif
 
 ifndef IPHONE_TARGET
-  $(error Please set IPHONE_TARGET to SIMULATOR or HARDWARE)
+  $(error Please set IPHONE_TARGET to SIMULATOR, M1_SIMULATOR, or HARDWARE)
```

Comment:
It seems like it would be easier for the user if we detected it. Like what they're talking about [here](https://stackoverflow.com/questions/65259300/detect-apple-silicon-from-command-line).

Also using the name "M1" in the NEWS is probably fine, but if we need a name for it here we should go with a different name. Going by [Apple's naming conventions for their processors](https://en.wikipedia.org/wiki/Apple_silicon#A_series) it's probably not going to be accurate to call it "M1" for long.

Reply:
Right, this should use the target instruction set architecture `arm64` instead of the name of any specific processor